akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -932,6 +932,26 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
   def grouped(n: Int): javadsl.Flow[In, java.util.List[Out], Mat] =
     new Flow(delegate.grouped(n).map(_.asJava)) // TODO optimize to one step
 
+  /**
+   * Chunk up this stream into groups of elements that have a cumulative weight greater than or equal to
+   * the `minWeight`, with the last group possibly smaller than requested `minWeight` due to end-of-stream.
+   *
+   * `minWeight` must be positive, otherwise IllegalArgumentException is thrown.
+   * `costFn` must return a non-negative result for all inputs, otherwise the stage will fail
+   * with an IllegalArgumentException.
+   *
+   * '''Emits when''' the cumulative weight of elements is greater than or equal to the `minWeight` or upstream completed
+   *
+   * '''Backpressures when''' a buffered group weighs more than `minWeight` and downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def groupedWeighted(minWeight: Long)(
+      costFn: function.Function[Out, java.lang.Long]): javadsl.Flow[In, java.util.List[Out], Mat] =
```

Comment:
Better use `java.util.Function` (or perhaps even `java.util.function.ToLongFunction`) here, the `akka.japi.function.Function` is for when throwing a checked except from the lambda would make sense and here it doesn't.

Reply:
Sure, I'll update it. For reference, I used that one because `groupedWeightedWithin` uses `akka.japi.function.Function` for the cost function. I'll update it to `java.util.Function`, as that is the other common pattern in the file.